### ISSUE TITLE
Allow the serializer specified in `NullsUsing` to access the serializer specified in `Using`

### ISSUE THREAD

**Issue by k163377** [FasterXML/jackson-databind#3744]
**Is your feature request related to a problem? Please describe.**
If the `JsonSerialize` annotation is set as in the sample code below, the `Using` serializer will not be obtained by executing `findValueSerializer` from `NullsUsing::serialize`.
Although not described in the sample code, the same is true when `NullsUsing` is executed from `AnnotationIntrospector`.

```java
import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;

public class Src {
    static class Using extends StdSerializer<Integer> {
        Using() { super(Integer.class); }

        @Override
        public void serialize(Integer value, JsonGenerator gen, SerializerProvider provider) throws IOException {
            gen.writeNumber(value + 100);
        }
    }

    static class NullsUsing extends StdSerializer<Integer> {
        NullsUsing() { super(Integer.class); }

        @Override
        public void serialize(Integer value, JsonGenerator gen, SerializerProvider provider) throws IOException {
            provider.findValueSerializer(Integer.class).serialize(-100, gen, provider);
        }
    }

    private Integer i;

    public Src(Integer i) { this.i = i; }

    @JsonSerialize(using = Using.class, nullsUsing = NullsUsing.class)
    public Integer getI() { return i; }

    public static void main(String[] args) throws JsonProcessingException {
        ObjectMapper mapper = new ObjectMapper();
        Src isNotNull = new Src(0);
        Src isNull = new Src(null);

        System.out.println(mapper.writeValueAsString(isNotNull)); // -> {"i":100}
        System.out.println(mapper.writeValueAsString(isNull)); // -> {"i":-100}
    }
}
```

**Describe the solution you'd like**
Change so that when `findValueSerializer` is called from a fallback such as `nullsUsing`, the serializer set to `using` is also retrieved.
(Just to confirm, this is not a bug but an intentional implementation, right?)

**Usage example**
I don't think this issue will cause fatal problems in general use cases.
However, I find it inconvenient that there is no way to call the serializer set in `JsonSerialize::using` if the property is `null` in a use case such as creating a module.

**Additional context**
This issue has been an obstacle in enhancing `value class` serialization support in `jackson-module-kotlin`/`jackson-module-kogera`.
I tried to implement the following policy, but the result of step 3 is not as expected because I cannot access the serializer set in `JsonSerialize::using`.

1. `Java Property` returns `null`
2. wrapping `null` into `value class` in the process in `nullSerializer`
3. serialize with the serializer corresponding to 2

**Comment by cowtowncoder** [comment]
Right, this is the way things work and it is unlikely to be changed: `findValueSerializer()` uses global (per-Mapper) definitions of serializers by type, whereas `@JsonSerialize(using=)` in this case just overrides serialized used for that accessor (and cannot be registered for that type as that could randomly override default serializer when different accessors had different serializers).

I can see why access to locally-overridden serializer would be useful in use cases like this, but I don't really see a viable way to do that -- and it seems like lots of work for something may be more of an edge case.
